### ISSUE TITLE
[Bug] Alert request to Prometheus under Basic Auth not working

### ISSUE THREAD

**Issue by hagen1778** [grafana/grafana#6799]
Please include this information:
- What Grafana version are you using? - grafana-4.0.0-1480439068
- What datasource are you using? - Prometheus
- What OS are you running grafana on? - Ubuntu 16.04
- What did you do? - Trying to setup some alerts
- What was the expected result? - Working test rule
- What happened instead? -  tsdb.HandleRequest() error bad_response: bad response code 401

I am trying to create simple alert:
![6ca641acd3384f7bb2fe0639fe8f44c4](https://cloud.githubusercontent.com/assets/2902918/20830512/7a3cd4cc-b889-11e6-8cfb-b7b4821905c7.png)

The query is:
`sum(go_goroutines)`

My Prometheus is protected by Basic Auth. Here is tcpdump of incoming request:
```
".f.^1z!GET /api/v1/query_range?end=2016-12-02T10%3A19%3A23.747797219Z&query=sum%28go_goroutines%29&start=2016-12-02T10%3A14%3A23.747797219Z&step=30.000 HTTP/1.1
Host: prometheus.domain.com
User-Agent: Go-http-client/1.1
Accept-Encoding: gzip
```

If i'll just refresh current dashboards - i'll receive next request:
```
"...^2.OGET /api/v1/query_range?query=sum(go_goroutines)&start=1480652661&end=1480674261&step=30 HTTP/1.1
Host: prometheus.domain.com
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2840.90 Safari/537.36
Accept: application/json, text/plain, */*
Accept-Encoding: gzip, deflate, sdch
Accept-Language: ru-RU,ru;q=0.8,en-US;q=0.6,en;q=0.4,nl;q=0.2,uk;q=0.2
Authorization: Basic <password>
Referer: http://grafana.domain.com/dashboard/db/test-alerts?panelId=1&fullscreen&edit&tab=alert
X-Forwarded-For: <forwarded>
X-Real-Ip: <realip>
```

Alert requests to Prometheus are without auth credentials, while data source is configured to do this.

**Comment by flyersa** [comment]
same as here https://github.com/grafana/grafana/issues/6784 with influxdb

**Comment by martindolecek** [comment]
same problem (alerts > prometheus with auth-basic = 401)

**Comment by hagen1778** [comment]
Problem still exists

**Comment by torkelo** [comment]
if anyone wants to help out the issue is here
https://github.com/grafana/grafana/blob/master/pkg/tsdb/prometheus/prometheus.go#L50

I guess the prometheus client as basic auth options that needs to be configured there.

**Comment by bergquist** [comment]
Closed by #7466
